datashader/glyphs.py

```diff
@@ -32,25 +31,49 @@ def validate(self, in_dshape):
     @staticmethod
     @ngjit
     def _compute_x_bounds(xs):
+        if len(xs) == 0:
+            raise ValueError('x coordinate array is empty.')
```

Comment:
In that case x < minval and x < maxval will both always be False, so you would only ever get NaN out.